### PR TITLE
Make EnumLookupError inherit from LookupError.

### PR DESCRIPTION
**Pull request by jemiahlee**

### CODE REVIEW

File: tests/richenum/test_ordered_rich_enums.py [hunk 1]
```diff
@@ -45,8 +45,9 @@ def test_lookup_by_index(self):
         # Should work if enum isn't zero-indexed.
         self.assertEqual(SadBreakfast.from_index(1), oatmeal)
 
-        with self.assertRaises(EnumLookupError):
+        with self.assertRaises(EnumLookupError) as error:
             SadBreakfast.from_index(7)
+        self.assertTrue(isinstance(error.exception, LookupError))
```
Comment (dhui) [inline tests/richenum/test_ordered_rich_enums.py]: This should be it's own test. e.g. a test for exception inheritance. Then we don't need to add addition assert checks in these unrelated tests.

Comment (tulioz) [inline tests/richenum/test_ordered_rich_enums.py]: Also can use: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertIsInstance

Comment (jemiahlee) [inline tests/richenum/test_ordered_rich_enums.py]: The disadvantage of not having those commit checks anymore.

File: tests/richenum/test_rich_enums.py [hunk 2]
```diff
@@ -58,6 +58,11 @@ def test_enums_iterate_through_members(self):
         members = set(e for e in Vegetable)
         self.assertEqual(members, set((Vegetable.OKRA, Vegetable.BROCCOLI)))
 
+    def test_lookup_error_type(self):
+        with self.assertRaises(EnumLookupError) as error:
+            Vegetable.from_canonical('parsnip')
+        self.assertIsInstance(error.exception, LookupError)
```
Comment (dhui) [inline tests/richenum/test_rich_enums.py]: You could change this whole test to just do:
`self.assertTrue(issubclass(EnumLookupError, LookupError))`

I think all of the other cases are covered by the `self.assertRaises(EnumLookupError)` check in other tests.

IIRC, we have other tests to ensure that each RichEnum specific  `LookupError` inherits from `EnumLookupError`

**Comment by jemiahlee** [comment]
@dhui updated the last commit

**Comment by dhui** [comment]
:shipit:

**Comment by jemiahlee** [comment]
I'll squash commits and re-push.

**Pull request merged**